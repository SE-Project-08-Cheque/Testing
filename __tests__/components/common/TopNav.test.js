// import dependencies
import React from "react";

// import react-testing methods
import { render, fireEvent, cleanup } from "@testing-library/react";

// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";

// import react dom
import ReactDOM from "react-dom";

// import renderer for take snapshots
import renderer from "react-test-renderer";

// import the component for testing
import TopNav from "../../../components/common/TopNav";

afterEach(cleanup);


it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <TopNav routes = {[]}/>,
    div
  );
});

it("matches snapshot", () => {
    const tree = renderer.create(
      <TopNav routes = {[]}/>
    );
  
    expect(tree).toMatchSnapshot();
  });
  