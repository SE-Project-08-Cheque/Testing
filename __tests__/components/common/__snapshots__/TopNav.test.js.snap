// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-pzpr4w"
>
  <div
    className="css-1td3zw5"
  >
    <div
      className="chakra-stack css-m662aq"
    >
      <h1
        className="chakra-heading css-gvp5wc"
      >
        SunRise
      </h1>
      <div
        className="chakra-stack css-1kjztpx"
      >
        <div
          className="chakra-stack css-1p1m4ay"
        />
      </div>
    </div>
  </div>
</div>
`;
